lib/skate_web/controllers/detour_missed_stops_controller.ex

```diff
@@ -0,0 +1,52 @@
+defmodule SkateWeb.DetourMissedStopsController do
+  use SkateWeb, :controller
+
+  alias Skate.Detours.MissedStops
+  alias Util.Location
+
+  @spec missed_stops(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def missed_stops(conn, %{
+        "route_pattern_id" => route_pattern_id,
+        "connection_start" => connection_start,
+        "connection_end" => connection_end
+      }) do
+    with route_pattern <- route_pattern(route_pattern_id),
+         false <- is_nil(route_pattern),
+         shape_with_stops <-
+           shape_with_stops(route_pattern.representative_trip_id),
+         false <- is_nil(shape_with_stops) do
+      connection_start_location = Location.new(connection_start["lat"], connection_start["lon"])
+      connection_end_location = Location.new(connection_end["lat"], connection_end["lon"])
+
+      missed_stops =
+        missed_stops(%MissedStops{
+          connection_start: connection_start_location,
+          connection_end: connection_end_location,
+          stops: shape_with_stops.stops,
+          shape: shape_with_stops.points
+        })
+
+      json(conn, %{data: missed_stops})
+    else
+      _ -> send_resp(conn, :bad_request, "bad request")
+    end
+  end
+
+  defp route_pattern(route_pattern_id) do
+    Application.get_env(:skate_web, :route_pattern_fn, &Schedule.route_pattern/1).(
+      route_pattern_id
+    )
+  end
+
+  defp shape_with_stops(trip_id) do
+    Application.get_env(
+      :skate_web,
+      :shape_with_stops_fn,
+      &Schedule.shape_with_stops_for_trip/1
+    ).(trip_id)
+  end
+
+  defp missed_stops(cfg) do
+    Application.get_env(:skate_web, :missed_stops_fn, &MissedStops.missed_stops/1).(cfg)
```

Comment:
```suggestion
  defp missed_stops(args) do
    Application.get_env(:skate_web, :missed_stops_fn, &MissedStops.missed_stops/1).(args)
```

**Nit**: I got confused at first and thought that `cfg` was some kind of application-level config.

Reply:
I borrowed the `cfg` terminology from the `MissedStops` module itself, though I agree that `args` is probably clearer. The `MissedStops` struct is a little weird IMO because it's only used as the argument to the `missed_stops` function but maybe @firestack could speak to why it's implemented that way?

Reply:
I probably should've used something that would accept either a keyword list or map, but  I didn't want to make it overly complex. I was finding that it was easy to get the args confused and it ended up being difficult to understand at a glance when reading the code back, so for readability's sake, I went with a struct. 

I'm fine with changing it, though I think the arguments should continue to be named in the calling code. I liked using a struct because it allows for requiring all fields to be present and making that somewhat visible at "writing time" and compile time.

Reply:
My perhaps controversial opinion is that I'd just write it as a `/4` function, though I think we should hold off on any refactoring on that and do it in another ticket / PR if we want to. In the meantime I like @joshlarson's naming change so I'll apply the suggestion.